zokrates_cli/src/ops/compile.rs

```diff
@@ -74,16 +80,23 @@ pub fn exec(sub_matches: &ArgMatches) -> Result<(), String> {
 }
 
 fn cli_compile<T: Field>(sub_matches: &ArgMatches) -> Result<(), String> {
-    println!("Compiling {}\n", sub_matches.value_of("input").unwrap());
+    // TODO: remove the warning once light flag is removed entirely
+    if sub_matches.is_present("light") {
```

Comment:
I suspect this is always false, as it get overridden

Reply:
It's actually reversed, `--light` overrides `--ztf` and `--verbose`

Reply:
I'm struggling to understand the mechanism here, maybe I need another coffee can you explain what happens in different situations? Also `overrides_with_all` that ordering matter, is that taken into account here?

Reply:
When you pass `--light` flag it overrides both `--ztf` and `--verbose` (order does not matter here), essentially as if they were never used.

so a scenario like the following
```
zokrates compile -i main.zok --ztf --verbose --light
```
will result in
```
zokrates compile -i main.zok --light
```

This is the case where a warning will be displayed. The `--light` flag doesn't do anything here because its behaviour is now by default, but if other conflicting flags are passed (`ztf`, `verbose`) it will override them to make sure the logic stays the same as it was before. The flag is also set as hidden (but its still usable and will display a warning of deprecation) to force people to use newer flags.

Some other examples can be found in the `clap` docs https://ruststudy.github.io/tao_of_rust_docs/structopt/clap/struct.Arg.html#method.overrides_with_all

Reply:
Okay that makes sense, so in the backward compatible case ordering doesn't matter because only `light` is there? Thanks!

Reply:
Exactly!